worker/worker.go

```diff
@@ -445,6 +462,177 @@ func (w *Worker) processClaim(claim taskClaim) {
+	util.Spawn(len(result.Supersedes), func(i int) {
+		taskID := result.Supersedes[i]
+		// Don't attempt to reclaim the initial task
+		if taskID == claim.Status.TaskID {
+			return
+		}
+		// Get state of the task, to find runID
+		s, qerr := w.queue.Status(taskID)
+		if qerr != nil {
+			m.WithTag("taskId", taskID).Infof("unable to get task status, error: %v", qerr)
+			return
+		}
+		runID := len(s.Status.Runs) - 1
+		if runID < 0 || s.Status.Runs[runID].State != "pending" {
+			return
+		}
+		c, qerr := w.queue.ClaimTask(taskID, strconv.Itoa(runID), &queue.TaskClaimRequest{
+			WorkerID:    w.options.WorkerID,
+			WorkerGroup: w.options.WorkerGroup,
+		})
+		if qerr != nil {
+			m.WithTags(map[string]string{
+				"taskId": taskID,
+				"runId":  strconv.Itoa(runID),
+			}).Debug("unable to claimTask from superseder, error: %v", qerr)
+			return
+		}
+		claimAttempts[i] = taskClaim(*c)
+	})
+
+	// remove invalid claims
+	claims := []taskClaim{claim}
+	for _, claim := range claimAttempts {
+		if claim.Status.TaskID != "" {
+			claims = append(claims, claim)
+		}
+	}
+
+	// Take the last claim we have, as the task we run
+	claim = claims[len(claims)-1]
+	claims = claims[:len(claims)-1]
+
+	// Start a reclaiming loop, and finish off by resolving superseded
+	var stopReclaiming atomics.Once
+	var tasksResolved atomics.WaitGroup
+	tasksResolved.Add(len(claims))
+	ctx, cancel := context.WithCancel(context.Background())
+	go util.Spawn(len(claims), func(i int) {
+		defer tasksResolved.Done()
+		taskID := claims[i].Status.TaskID
+		runID := strconv.Itoa(claims[i].RunID)
+		for {
+			select {
+			case <-time.After(10 * time.Minute):
+				// reclaim claims[i]
+				q := w.newQueueClient(ctx, asClientCredentials(claims[i].Credentials))
+				result, qerr := q.ReclaimTask(taskID, runID)
+				if qerr != nil {
+					m.WithTags(map[string]string{
+						"taskId": taskID,
+						"runId":  runID,
+					}).Warnf("failed reclaimTask error: %v", qerr)
+					return
+				}
+				claims[i].Credentials = result.Credentials
+				claims[i].TakenUntil = result.TakenUntil
+			case <-stopReclaiming.Done():
+				// resolve claims[i] as superseded
+				q := w.newQueueClient(ctx, asClientCredentials(claims[i].Credentials))
+				// TODO: Upload artifacts
+				_, qerr := q.ReportException(taskID, runID, &queue.TaskExceptionRequest{
+					Reason: "superseded",
+				})
+				if qerr != nil {
+					m.WithTags(map[string]string{
+						"taskId": taskID,
+						"runId":  runID,
+					}).Warnf("failed to reportException with reason superseded, error: %v", qerr)
+				}
+				return
+			}
+		}
+	})
+
+	// Remove supersederUrl from payload, so that it doesn't create malformed-payload
+	payload = make(map[string]interface{})
+	if json.Unmarshal(claim.Task.Payload, &payload) == nil {
+		delete(payload, "supersederUrl")
+		claim.Task.Payload, err = json.Marshal(payload)
+		if err != nil {
+			panic(errors.Wrap(err, "failed to serialize data known to be JSON"))
+		}
+	}
```

Comment:
Yeah, I considered it.... but it just seems overkill.. This way I get to keep all the ugly superseded stuff in a single method rather than creating a lot of ugly functions...

Reply:
Fair enough.

Reply:
> Yeah, I considered it.... but it just seems overkill.. This way I get to keep all the ugly superseded stuff in a single method rather than creating a lot of ugly functions...

A [function literal](https://golang.org/ref/spec#Function_literals) would solve that.

Reply:
> A function literal would solve that.

Yes, again, I don't care... the less is contained in a single function we can easily delete.